BlueprintUI/Sources/Element/UIViewElement.swift

```diff
@@ -0,0 +1,156 @@
+//
+//  UIViewElement.swift
+//  BlueprintUI
+//
+//  Created by Kyle Van Essen on 6/4/20.
+//
+
+import UIKit
+
+
+///
+/// An element type  which makes it easier to wrap an existing `UIView` instance that
+/// provides its own sizing via `sizeThatFits`. An instance of the view is used for
+/// sizing and measurement, so that you do not need to re-implement your own measurement.
+///
+/// Example
+/// -------
+/// If you were implementing a very basic `Label` element, your implementation would look something
+/// like this:
+/// ```
+/// struct Label : UIViewElement {
+///
+///     var text : String
+///     var font : UIFont
+///
+///     typealias UIViewType = UILabel
+///
+///     static func makeUIView() -> UILabel {
+///         UILabel()
+///     }
+///
+///     func updateUIView(_ view: UILabel) {
+///         view.text = self.text
+///         view.font = self.font
+///     }
+/// }
+/// ```
+public protocol UIViewElement : Element {
+    
+    /// The type of the view associated with the element.
+    associatedtype UIViewType : UIView
+        
+    /// Create and return a new instance of the provided view type.
+    ///
+    /// Note
+    /// ----
+    /// Ensure that you do not pass any values to the initializer of your view type
+    /// that you cannot also update in `updateUIView(_:)`, as view instances
+    /// are reused for sizing and measurement.
+    static func makeUIView() -> UIViewType
```

Comment:
Instead of creating a new abstraction for making/updating a view, why not use ViewDescription?

Reply:
Wanted to match the SwiftUI API, which IMO is a lot clearer and more concise to write.

Reply:
As well, we wouldnt have view description (likely) if we did not have to type erase to element – a lot of the value of ViewDescription is the type safety of the creation and usage of the view. Since we can use an associated type here, it's clearer to just have two methods.